### PR TITLE
🐛 Destinations v2: handle streams with no columns

### LINKED ISSUE #29090: BigQuery Destination V2 - Number of arguments does not match for function ARRAY_CONCAT
### Connector Name

destination-bigquery

### Connector Version

1.7.2

### What step the error happened?

During the sync

### Revelant information

Normalization: raw data (json)
Source: mongodb

### Relevant log output

```shell
Stack Trace: com.google.cloud.bigquery.BigQueryException: Query error: Number of arguments does not match for function ARRAY_CONCAT. Supported signature: ARRAY_CONCAT(ARRAY, [ARRAY, ...]) at [17:3]
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:114)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getQueryResults(HttpBigQueryRpc.java:694)
	at com.google.cloud.bigquery.BigQueryImpl$36.call(BigQueryImpl.java:1437)
	at com.google.cloud.bigquery.BigQueryImpl$36.call(BigQueryImpl.java:1432)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.bigquery.BigQueryRetryHelper.run(BigQueryRetryHelper.java:86)
	at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:49)
	at com.google.cloud.bigquery.BigQueryImpl.getQueryResults(BigQueryImpl.java:1431)
	at com.google.cloud.bigquery.BigQueryImpl.getQueryResults(BigQueryImpl.java:1415)
	at com.google.cloud.bigquery.Job$1.call(Job.java:338)
	at com.google.cloud.bigquery.Job$1.call(Job.java:335)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.bigquery.BigQueryRetryHelper.run(BigQueryRetryHelper.java:86)
	at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:49)
	at com.google.cloud.bigquery.Job.waitForQueryResults(Job.java:334)
	at com.google.cloud.bigquery.Job.waitFor(Job.java:244)
	at io.airbyte.integrations.destination.bigquery.typing_deduping.BigQueryDestinationHandler.execute(BigQueryDestinationHandler.java:63)
	at io.airbyte.integrations.base.destination.typing_deduping.DefaultTyperDeduper.typeAndDedupe(DefaultTyperDeduper.java:100)
	at io.airbyte.integrations.destination.bigquery.BigQueryStagingConsumerFactory.lambda$onCloseFunction$6(BigQueryStagingConsumerFactory.java:243)
	at io.airbyte.integrations.destination.buffered_stream_consumer.BufferedStreamConsumer.close(BufferedStreamConsumer.java:306)
	at io.airbyte.integrations.base.FailureTrackingAirbyteMessageConsumer.close(FailureTrackingAirbyteMessageConsumer.java:82)
	at io.airbyte.integrations.base.Destination$ShimToSerializedAirbyteMessageConsumer.close(Destination.java:95)
	at io.airbyte.integrations.base.IntegrationRunner.lambda$runInternal$0(IntegrationRunner.java:154)
	at io.airbyte.integrations.base.IntegrationRunner.watchForOrphanThreads(IntegrationRunner.java:272)
	at io.airbyte.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:157)
	at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:99)
	at io.airbyte.integrations.destination.bigquery.BigQueryDestination.main(BigQueryDestination.java:455)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
GET https://www.googleapis.com/bigquery/v2/projects/obf-project/queries/job_VVD1tFyAT-iDvP8vq7LlvcJHxbTg?location=EU&maxResults=0&prettyPrint=false
{
  "code": 400,
  "errors": [
    {
      "domain": "global",
      "location": "q",
      "locationType": "parameter",
      "message": "Query error: Number of arguments does not match for function ARRAY_CONCAT. Supported signature: ARRAY_CONCAT(ARRAY, [ARRAY, ...]) at [17:3]",
      "reason": "invalidQuery"
    }
  ],
  "message": "Query error: Number of arguments does not match for function ARRAY_CONCAT. Supported signature: ARRAY_CONCAT(ARRAY, [ARRAY, ...]) at [17:3]",
  "status": "INVALID_ARGUMENT"
}
	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:146)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:118)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:37)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:428)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1111)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:514)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:455)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:565)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getQueryResults(HttpBigQueryRpc.java:692)
	... 25 more
```

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte/issues/29090

add a sqlgenerator test case; make some very minor tweaks to the sqlgenerators to support this.

### CODE REVIEW

File: airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java [hunk 1]
```diff
@@ -718,20 +718,58 @@ public void weirdColumnNames() throws Exception {
...
+  public void noColumns() throws Exception {
+    createRawTable(streamId);
+    insertRawTableRecords(
+        streamId,
+        List.of(Jsons.deserialize(
+            """
+                {
+                  "_airbyte_raw_id": "14ba7c7f-e398-4e69-ac22-28d578400dbc",
+                  "_airbyte_extracted_at": "2023-01-01T00:00:00Z",
+                  "_airbyte_data": {}
```
Comment (evantahler) [inline airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java]: would _airbyte_data be null in this case? I have no idea

**Review by evantahler — APPROVED** [review]
If the new tests are passing, 👍 from me!

File: airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java]: :shrug:

my theory is that the data actually has random columns that aren't declared in the schema. And the protocol declares `data` as required, so I think we're ok to assume a nonnull object https://github.com/airbytehq/airbyte-protocol/blob/main/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L55-L61

Comment (edgao) [reply airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java]: also we're currently declaring `_airbyte_data` as `not null` on the raw tables >.>

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`463c8fa23f`](https://github.com/airbytehq/airbyte/pull/29381/commits/463c8fa23fba3a3e908434cfff3c0533c3119afe)) - ❌

⏲️  Total pipeline duration: 47mn19s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1691792386/463c8fa23fba3a3e908434cfff3c0533c3119afe/destination-snowflake/1.3.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:463c8fa23fba3a3e908434cfff3c0533c3119afe)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`584fae5d92`](https://github.com/airbytehq/airbyte/pull/29381/commits/584fae5d9262a0e4859001a196a2ae2a6b3a8879)) - ✅

⏲️  Total pipeline duration: 31mn09s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1691795697/584fae5d9262a0e4859001a196a2ae2a6b3a8879/destination-bigquery/1.7.7/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:584fae5d9262a0e4859001a196a2ae2a6b3a8879)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`584fae5d92`](https://github.com/airbytehq/airbyte/pull/29381/commits/584fae5d9262a0e4859001a196a2ae2a6b3a8879)) - ❌

⏲️  Total pipeline duration: 47mn02s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1691795697/584fae5d9262a0e4859001a196a2ae2a6b3a8879/destination-snowflake/1.3.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:584fae5d9262a0e4859001a196a2ae2a6b3a8879)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`cc41964c79`](https://github.com/airbytehq/airbyte/pull/29381/commits/cc41964c79743d13d6e28c0496efe584999e3c8a)) - ✅

⏲️  Total pipeline duration: 29mn16s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1692115406/cc41964c79743d13d6e28c0496efe584999e3c8a/destination-bigquery/1.7.7/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cc41964c79743d13d6e28c0496efe584999e3c8a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`cc41964c79`](https://github.com/airbytehq/airbyte/pull/29381/commits/cc41964c79743d13d6e28c0496efe584999e3c8a)) - ❌

⏲️  Total pipeline duration: 43mn30s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1692115406/cc41964c79743d13d6e28c0496efe584999e3c8a/destination-snowflake/1.3.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cc41964c79743d13d6e28c0496efe584999e3c8a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Review by jbfbell — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`3e5e788dfb`](https://github.com/airbytehq/airbyte/pull/29381/commits/3e5e788dfb5509dc131f2ab4d259d9a2e7bfbefc)) - ✅

⏲️  Total pipeline duration: 45mn34s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1692227089/3e5e788dfb5509dc131f2ab4d259d9a2e7bfbefc/destination-snowflake/1.3.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3e5e788dfb5509dc131f2ab4d259d9a2e7bfbefc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`3e5e788dfb`](https://github.com/airbytehq/airbyte/pull/29381/commits/3e5e788dfb5509dc131f2ab4d259d9a2e7bfbefc)) - ✅

⏲️  Total pipeline duration: 33mn08s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_1s1t_no_columns/1692227089/3e5e788dfb5509dc131f2ab4d259d9a2e7bfbefc/destination-bigquery/1.7.7/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3e5e788dfb5509dc131f2ab4d259d9a2e7bfbefc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Pull request merged**